src/dtclient/kubernetes_settings.go

```diff
@@ -133,6 +118,57 @@ func (dtc *dynatraceClient) CreateOrUpdateKubernetesSetting(clusterLabel, kubeSy
 	return resDataJson[0].ObjectId, nil
 }
 
+func (dtc *dynatraceClient) getKubernetesSettingBody(clusterLabel, kubeSystemUUID, scope string, schemaId string, isK8sHierarchicalMonitoringSettings bool) []postKubernetesSettingsBody {
+	currentSchemaVersion := defaultSchemaVersion
+	if isK8sHierarchicalMonitoringSettings {
+		currentSchemaVersion = hierarchicalMonitoringSettingsSchemaVersion
+	}
+	body := []postKubernetesSettingsBody{
+		{
+			SchemaId:      schemaId,
+			SchemaVersion: currentSchemaVersion,
+			Value: postKubernetesSettings{
+				Enabled:          true,
+				Label:            clusterLabel,
+				ClusterIdEnabled: true,
+				ClusterId:        kubeSystemUUID,
+			},
+		},
+	}
+	if scope != "" {
+		body[0].Scope = scope
+	}
+	if !isK8sHierarchicalMonitoringSettings {
+		ms := MonitoringSettings{
+			CloudApplicationPipelineEnabled: true,
+			OpenMetricsPipelineEnabled:      false,
+			EventProcessingActive:           false,
+			FilterEvents:                    false,
+			EventProcessingV2Active:         false,
+		}
+
+		body[0].Value.MonitoringSettings = &ms
+	}
+	return body
+}
```

Comment:
I would split it this up a bit to be nicer:

We should only work with a list in the end so we don't do stuff like `body[0]...`

So would have a function for creating the base settings body: 
```
func createBaseKubernetesSettings(... ) postKubernetesSettingsBody {
	base := postKubernetesSettingsBody{
			SchemaId:      schemaId,
			Value: postKubernetesSettings{
				Enabled:          true,
				Label:            clusterLabel,
				ClusterIdEnabled: true,
				ClusterId:        kubeSystemUUID,
			},
		}
	if scope != "" {
		base.Scope = scope
	}
}
```

then have separate functions for the new Schema and the old Schema:
```
func createV3KubernetesSettingsBody(...) []postKubernetesSettingsBody {
	settings := createBaseKubernetesSettings(...)
	settings.Schema = hierarchicalMonitoringSettingsSchemaVersion
	ms := MonitoringSettings{
			CloudApplicationPipelineEnabled: true,
			OpenMetricsPipelineEnabled:      false,
			EventProcessingActive:           false,
			FilterEvents:                    false,
			EventProcessingV2Active:         false,
	}
	settings.Value.MonitoringSettings = &ms
	return []postKubernetesSettingsBody{settings}
}

func createV1KubernetesSettingsBody(...) []postKubernetesSettingsBody {
	settings := createBaseKubernetesSettings(...)
	settings.Schema = defaultSchemaVersion
	return []postKubernetesSettingsBody{settings}
}
```
by doing this you remove the weird ifs + bool logic, so you can make the code more readable and clear what it does

and the following reads a bit nicer:
```
func (dtc *dynatraceClient) CreateOrUpdateKubernetesSetting(clusterLabel, kubeSystemUUID, scope string) (string, error) {
	if kubeSystemUUID == "" {
		return "", errors.New("no kube-system namespace UUID given")
	}
	body := createV3KubernetesSettingsBody(clusterLabel, kubeSystemUUID, scope, schemaId)
	objectId, err := dtc.performCreateOrUpdateKubernetesSetting(body)
	if err != nil {
		if strings.Contains(err.Error(), strconv.Itoa(http.StatusNotFound)) {
			body = createV1KubernetesSettingsBody(clusterLabel, kubeSystemUUID, scope, schemaId)
			return dtc.performCreateOrUpdateKubernetesSetting(body)
		} else {
			return "", err
		}
	}
	return objectId, nil
}
```

Reply:
thanks, moved it to code